go/vt/wrangler/vexec.go

```diff
@@ -231,6 +239,34 @@ func (vx *vexec) exec() (map[*topo.TabletInfo]*querypb.QueryResult, error) {
 	return results, allErrors.AggrError(vterrors.Aggregate)
 }
 
+// execCallback runs the provided callback function on backend shard primaries.
+// It collects query results from all shards and returns an aggregate (UNION
+// ALL -like) result.
+func (vx *vexec) execCallback(callback func(context.Context, *topo.TabletInfo) (*querypb.QueryResult, error)) (map[*topo.TabletInfo]*querypb.QueryResult, error) {
+	var wg sync.WaitGroup
+	allErrors := &concurrency.AllErrorRecorder{}
+	results := make(map[*topo.TabletInfo]*querypb.QueryResult)
+	var mu sync.Mutex
+	ctx, cancel := context.WithTimeout(vx.ctx, 10*time.Second)
+	defer cancel()
+	for _, primary := range vx.primaries {
+		wg.Add(1)
+		go func(ctx context.Context, primary *topo.TabletInfo) {
+			defer wg.Done()
+			qr, err := callback(ctx, primary)
+			if err != nil {
+				allErrors.RecordError(err)
+			} else {
+				mu.Lock()
+				results[primary] = qr
+				mu.Unlock()
```

Comment:
This is a complete nit, so at your discretion, but I like the idiomatic way of
```go
m.Lock()
defer m.Unlock()
// do something
```
Which makes it obvious that `Unlock()` wasn't forgotten. Obviously the above is just trivial code and no real problem. I just like to stick to the idiom.